common/utils/routes.ts

```diff
@@ -192,7 +193,11 @@ export function decodeQuery<R>(query: ParsedUrlQuery, codecMap: CodecMap): R {
 export function encodeQuery<T>(props: T, codecMap: CodecMap): ParsedUrlQuery {
   const map = new Map();
   for (const prop in codecMap) {
-    map.set(prop, codecMap[prop].encode(props[prop]));
+    const value = codecMap[prop].encode(props[prop]);
+
+    if (isNotUndefined(value)) {
+      map.set(prop, value);
+    }
```

Comment:
This is used by the ImageLink/ImagesLink/ItemLink/WorksLink plugins, and I decided to commonise it here so we could simplify the downstream components.